packages/dev/serializers/src/USDZ/USDZSerializer.ts

```diff
@@ -0,0 +1,732 @@
+        prepend references = @./geometries/Geometry_${mesh.uniqueId}.usda@</Geometry>
+        prepend apiSchemas = ["MaterialBindingAPI"]
+    )
+    {
+        matrix4d xformOp:transform = ${transform}
+        uniform token[] xformOpOrder = ["xformOp:transform"]
+        rel material:binding = </Materials/Material_${material.uniqueId}>
+    }
+    `;
+    }
+
+    private static _BuildMatrix(matrix: Matrix) {
+        const array = matrix.toArray() as number[];
+        return `( ${USDZExport._BuildMatrixRow(array, 0)}, ${USDZExport._BuildMatrixRow(array, 4)}, ${USDZExport._BuildMatrixRow(array, 8)}, ${USDZExport._BuildMatrixRow(
+            array,
+            12
+        )} )`;
+    }
+
+    private static _BuildMatrixRow(array: number[], offset: number) {
+        return `(${array[offset + 0]}, ${array[offset + 1]}, ${array[offset + 2]}, ${array[offset + 3]})`;
+    }
+
+    private static _BuildCamera(camera: Camera) {
+        const scene = camera.getScene();
+        const name = `${camera.name}_${camera.uniqueId}`;
+        const mat = camera.getWorldMatrix();
+
+        const transform = USDZExport._BuildMatrix(mat);
+
+        if (mat.determinant() < 0) {
+            Tools.Warn(`USDZExport: USDZ does not support negative scales on camera: ${camera.name}`);
+        }
+
+        if (camera.mode === Camera.ORTHOGRAPHIC_CAMERA) {
+            return `def Camera "${name}"
+                {
+                    matrix4d xformOp:transform = ${transform}
+                    uniform token[] xformOpOrder = ["xformOp:transform"]
+                    float2 clippingRange = (${camera.minZ.toPrecision(USDZExport.Precision)}, ${camera.maxZ.toPrecision(USDZExport.Precision)})
+                    float horizontalAperture = ${((Math.abs(camera.orthoLeft ?? 0) + Math.abs(camera.orthoRight ?? 0)) * 10).toPrecision(USDZExport.Precision)}
+                    float verticalAperture = ${((Math.abs(camera.orthoTop ?? 0) + Math.abs(camera.orthoBottom ?? 0)) * 10).toPrecision(USDZExport.Precision)}
+                    token projection = "orthographic"
+                }`;
+        } else {
+            return `def Camera "${name}"
+                {
+                    matrix4d xformOp:transform = ${transform}
+                    uniform token[] xformOpOrder = ["xformOp:transform"]
+                    float2 clippingRange = (${camera.minZ.toPrecision(USDZExport.Precision)}, ${camera.maxZ.toPrecision(USDZExport.Precision)})
+                    float focalLength = ${(USDZExport.FilmGauge * Math.min(scene.getEngine().getAspectRatio(camera, true), 1)).toPrecision(USDZExport.Precision)}
+                    float focusDistance = ${USDZExport.Focus.toPrecision(USDZExport.Precision)}
+                    float horizontalAperture = ${USDZExport.FilmGauge.toPrecision(USDZExport.Precision)} 
+                    token projection = "perspective"
+                    float verticalAperture = ${USDZExport.FilmGauge.toPrecision(USDZExport.Precision)}

```

Comment:
The values for `focalLength`, `horizontalAperture` and `verticalAperture` are still wrong.

https://github.com/mrdoob/three.js/blob/dev/examples/jsm/exporters/USDZExporter.js#L698-L702

https://github.com/mrdoob/three.js/blob/dev/src/cameras/PerspectiveCamera.js#L73-L100

Reply:
![image](https://github.com/BabylonJS/Babylon.js/assets/1538857/8e09ab02-115a-45fb-ba58-1bec141a3202)
No they are not, they are using the exact same numbers as the three deployment.    I'm not sure what you are looking at?

Reply:
I'm pretty sure we have no reason to support aspect calculations as we don't have any film gauge settings.

Reply:
From Threejs code:

USDZ exporter:
```javascript
float focalLength = ${ camera.getFocalLength().toPrecision( PRECISION ) }
```
`PerspectiveCamera` class:
```javascript
getFocalLength() {
	const vExtentSlope = Math.tan( MathUtils.DEG2RAD * 0.5 * this.fov );
	return 0.5 * this.getFilmHeight() / vExtentSlope;
}

getFilmHeight() {
	return this.filmGauge / Math.max( this.aspect, 1 );
}
```
So, in Babylon, `focalLength` would be calculated as:
```typescript
(0.5 * USDZExport.FilmGauge / Math.max(engine.getAspectRatio(camera, true), 1)) / Math.tan(0.5 * camera.fov)
```
Am I missing something?

> I'm pretty sure we have no reason to support aspect calculations as we don't have any film gauge settings.

Maybe, but since we're porting, let's just use the same code (I don't think any of us are willing to dig into the specification to understand where these parameters come from / how they should be calculated, whereas the guy(s) who wrote the Threejs code probably did).